Components/Core/Inc/Timer.hpp

```diff
@@ -25,21 +35,32 @@ constexpr uint32_t DEFAULT_TIMER_COMMAND_WAIT_PERIOD = MS_TO_TICKS(15); // Defau
 class Timer
 {
 public:
-	Timer();
-
-	bool ChangePeriod(const uint32_t period);
-
-	// WORK-IN-PROGRESS
-	// NOTES:
-	// - I can think of several timer types
-	// 1) Default Ctor Timer (1 second polling timer that requires polling to acquire state with no callback)
-	// 2) Callback Enabled Timer (user-provided callback)
+	Timer(); // Default Constructor (Polling Timer)
+	Timer(void (*TimerCallbackFunction_t)( TimerHandle_t xTimer )); // Constructor for Callback Enabled Timer
+	~Timer();
+	bool ChangePeriod(const uint32_t period_ms);
+	bool ChangePeriodAndStart(const uint32_t period_ms);
+	bool Start();
+	bool Stop();
+	bool ResetTimer();
+	bool ResetTimerAndStart();
+	void SetAutoReload(bool setReloadOn);
 
+	bool GetAutoReload();
+	TimerState GetState();
+	const uint32_t GetPeriod();
+	const uint32_t GetRemainingTime();
 
+	static void CallbackFunction( TimerHandle_t xTimer );
 
 protected:
 	TimerHandle_t rtTimerHandle;
-
+	TimerState timerState;
+	uint32_t remainingTime;
+	uint32_t remainingTimeBetweenPauses;
+	uint32_t rtosTimeRemaning();
```

Comment:
Spelling, also please add a few lines between this function and the variables, ideally put the function at the top, also please rename the function with a capital, like this:

```
	const uint32_t GetRTOSTimeRemaining() const;

	TimerHandle_t rtTimerHandle;
	TimerState timerState;
	uint32_t remainingTime;
	uint32_t remainingTimeBetweenPauses;
```